File: ChamberHero.Frontend/src/types/appointment.ts
import type { Patient } from "@/types/patient";

export type AppointmentStatus = "Waiting" | "InConsultation" | "Completed" | "Cancelled";

export interface Appointment {
  id: string;
  doctor_id: string;
  patient_id: string;
  chamber_id: string;
  serialNo: number;
  appointment_date: string;
  status: AppointmentStatus;
  patient: Patient;
  created_at?: string;
  updated_at?: string;
}

export interface AppointmentCheckInPayload {
  patientId: string;
  chamberId: string;
}

export interface AppointmentStatusUpdatePayload {
  status: AppointmentStatus;
}

File: ChamberHero.Backend/ChamberHero.Core/Entities/Appointment.cs
using System;

namespace ChamberHero.Core.Entities;

public class Appointment
{
    public Guid Id { get; set; }
    public Guid DoctorId { get; set; }
    public Guid PatientId { get; set; }
    public Guid ChamberId { get; set; }
    public int SerialNo { get; set; }
    public DateTime AppointmentDate { get; set; }
    public AppointmentStatus Status { get; set; } = AppointmentStatus.Waiting;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public Patient? Patient { get; set; }
    public Chamber? Chamber { get; set; }
}
